processor/transformer/transformer.go

```diff
@@ -530,3 +540,20 @@ func (trans *handle) userTransformURL() string {
 func (trans *handle) trackingPlanValidationURL() string {
 	return trans.config.destTransformationURL + "/v0/validate"
 }
+
+func trackLongRunningTransformation(ctx context.Context, stage string, tags map[string]string, timeout time.Duration, log logger.Logger) {
+	start := time.Now()
+	t := time.NewTimer(timeout)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			keysAndValues := []interface{}{"stage", stage, "duration", time.Since(start).String()}
+			for k, v := range tags {
+				keysAndValues = append(keysAndValues, k, v)
+			}
+			log.Errorw("Long running transformation detected", keysAndValues...)
```

Comment:
```suggestion
```suggestion
			log.Errorw("Long running transformation detected", 
			"stage", stage, 
			"duration", time.Since(start).String(), 
			keysAndValues...,
			)
```
```

Reply:
```
too many arguments in call to log.Errorw
	have (string, string, string, string, string, []interface{})
	want (string, ...interface{})
```

Reply:
my bad, I proposed another way that simplifies function signature as well.